### PR TITLE
replace QT uint by uint32_t in livevis tool

### PR DESCRIPTION
**Pull request by BenjaminW3**
Follow up on #1007.

### CODE REVIEW

**Comment by ax3l** [comment]
for the topic of the PR: this tools does not use CUDA. I think it's the standard `uint` (without clear define of bit-ness).

**Comment by BenjaminW3** [comment]
There is no ```uint``` in any C++ or C standard so this is most probably system defined and is not portable.

**Comment by psychocoderHPC** [comment]
It looks like a typedef in `QT`

/usr/include/qt5/QtCore/qglobal.h:
```
typedef unsigned int uint;
```

**Comment by BenjaminW3** [comment]
Ok. Should we leave this as is then?

**Comment by psychocoderHPC** [comment]
@BenjaminW3 : Yes we should, because this is QT code, therefore the usage of `uint` is fine but not nice.

IMO

**Comment by ax3l** [comment]
why should this be good? I think we should change it if QT is not expecting exactly that in an interface.

**Comment by ax3l** [comment]
hm [QImage::setPixel](http://doc.qt.io/qt-4.8/qimage.html#setPixel) might be an other story.

File: src/tools/livevis/client/glwidget.cpp [hunk 1]
```diff
@@ -48,7 +48,7 @@ GLWidget::~GLWidget()
     }
 }
 
-void GLWidget::update_image(uint width, uint height, const void * pixel_data)
+void GLWidget::update_image(uint32_t width, uint32_t height, const void * pixel_data)
```
Comment (ax3l) [inline src/tools/livevis/client/glwidget.cpp]: change is fine, called in `mainwindow.cpp`

File: src/tools/livevis/client/mainwindow.cpp [hunk 2]
```diff
@@ -112,7 +112,7 @@ MainWindow::MainWindow(QWidget *parent) :
     connect(m_simulationGallery, SIGNAL(connect_to(QString)), this, SLOT(connectToURI(QString)));
 
     /// enable steering of simulation by interaction with mouse and keyboard
-    connect(ui->centralWidget, SIGNAL(send_message(uint,uint,const void*)), this, SLOT(sendMessage(uint,uint,const void*)));
+    connect(ui->centralWidget, SIGNAL(send_message(uint32_t,uint32_t,const void*)), this, SLOT(sendMessage(uint32_t,uint32_t,const void*)));
```
Comment (ax3l) [inline src/tools/livevis/client/mainwindow.cpp]: needs an update of the user defined call in `glwidget.h`. please check for the alias `unsigned int`, too.

**Comment by psychocoderHPC** [comment]
Yes it is a regular type in QT. It is not nice that QT uses types where the size is not defined but we had no advantage if we change it.

**Comment by ax3l** [comment]
but that only affects the [QImage::setPixel](http://doc.qt.io/qt-4.8/qimage.html#setPixel) call. user-defined slots can be changed to be more explicit.

For both this PR and the PIConGPU changes: let us pls check and correct the alias such as `unsigned int`, too.

File: src/tools/livevis/client/simulationgallery.cpp [hunk 3]
```diff
@@ -64,7 +64,7 @@ void SimulationGallery::add_vis_name_uri_thumb(QString name, QString uri/*, QIma
         for (int y = 0; y < 200; y++)
         {
             if ((x < 10 || x > 389) || (y < 4 || y > 195))
-                image.setPixel(x, y, std::numeric_limits<uint>::max() - 1);
+                image.setPixel(x, y, std::numeric_limits<uint32_t>::max() - 1);
```
Comment (ax3l) [inline src/tools/livevis/client/simulationgallery.cpp]: might need a revert, [QImage::setPixel](http://doc.qt.io/qt-4.8/qimage.html#setPixel)

**Comment by BenjaminW3** [comment]
Due to the ```uint``` being defined by Qt and the ```unsigned int``` in the interfaces being inherited from rivlib, I am for keeping it as is for now.

**Pull request closed without merging**